tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/injected/ManagedThreadFactoryDefinitionOnEJBInjectedServlet.java

```diff
@@ -0,0 +1,247 @@
+            IntContext.set(0);
+            StringContext.set("");
+        }
+    }
+
+    /**
+     * ManagedThreadFactoryDefinition defined on an EJB with minimal attributes
+     * creates threads with normal priority and uses java:comp/DefaultContextService
+     * to determine context propagation and clearing.
+     */
+    public void testManagedThreadFactoryDefinitionDefaultsEJB() throws Throwable {
+        CountDownLatch blocker = new CountDownLatch(1);
+        CountDownLatch allThreadsRunning = new CountDownLatch(2);
+        CompletableFuture<Object> lookupTaskResult = new CompletableFuture<>();
+        CompletableFuture<Object> txTaskResult = new CompletableFuture<>();
+
+        Runnable lookupTask = () -> {
+            try {
+                allThreadsRunning.countDown();
+                blocker.await(MAX_WAIT_SECONDS * 5, TimeUnit.SECONDS);
+                lookupTaskResult.complete(threadFactoryB);
+            } catch (Throwable x) {
+                lookupTaskResult.completeExceptionally(x);
+            }
+        };
+
+        Runnable txTask = () -> {
+            try {
+                allThreadsRunning.countDown();
+                UserTransaction trans = InitialContext.doLookup("java:comp/UserTransaction");
+                int initialStatus = trans.getStatus();
+                trans.begin();
+                try {
+                    blocker.await(MAX_WAIT_SECONDS * 5, TimeUnit.SECONDS);
+                } finally {
+                    trans.rollback();
+                }
+                txTaskResult.complete(initialStatus);
+            } catch (Throwable x) {
+                txTaskResult.completeExceptionally(x);
+            }
+        };
+
+        try {
+            threadFactoryB.newThread(lookupTask).start();
+            threadFactoryB.newThread(txTask).start();
+
+            assertTrue(allThreadsRunning.await(MAX_WAIT_SECONDS, TimeUnit.SECONDS),
+                    "ManagedThreadFactory threads must start running.");
+
+            blocker.countDown();
+
+            Object result;
+            
+            result = txTaskResult.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+            if (result instanceof Throwable) {
+                throw new AssertionError().initCause((Throwable) result);
+            }
+            assertEquals(result, Status.STATUS_NO_TRANSACTION,
+                    "Transaction context must be cleared from async Callable task "
+                            + "per java:comp/concurrent/EJBThreadFactoryB configuration.");
+
+            result = lookupTaskResult.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+            if (result instanceof Throwable)
+                throw new AssertionError().initCause((Throwable) result);
+            assertTrue(result instanceof ManagedThreadFactory, "Application context must be propagated to first thread "
+                    + "per java:comp/concurrent/EJBThreadFactoryB configuration.");
+        } finally {
+            IntContext.set(0);
```

Comment:
You can remove this if you want to. The test doesn't use IntContext so it shouldn't need to be cleared.
```suggestion
            IntContext.set(0);
```